llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -3276,8 +3308,9 @@ class BoUpSLP {
                           SmallVectorImpl<Value *> *AltScalars = nullptr) const;
 
     /// Return true if this is a non-power-of-2 node.
-    bool isNonPowOf2Vec() const {
-      bool IsNonPowerOf2 = !has_single_bit(Scalars.size());
+    bool isNonPowOf2Vec(const TargetTransformInfo &TTI) const {
```

Comment:
Style wise, I object to this particularly part of the change.  Having a vector which can be split into power-of-two pieces (which I think is what you're doing here?), is not the same as that vector *not being a power of two*.  You can adjust the callers to allow this special case if you want, but please revert the API change here.  It makes the code markedly less readable.

Reply:
This whole non-power-of-2 breaks everything and I said it before. This is a temporary solution, so I don't see an issue here. It will be removed completely once full non-power-of-2 support is landed

Reply:
The quality of intermediate code states matter.  I am specifically providing feedback as a reviewer, and asking you to make a style change.  Please do so.

Reply:
I disagree